extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/IngressTlsConfig.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.kubernetes.deployment;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class IngressTlsConfig {
+
+    /**
+     * If true, it will use the TLS configuration in the generated Ingress resource.
+     */
+    @ConfigItem
+    boolean enabled;
```

Comment:
I don't really like having to add this `enabled` field... However, I didn't see any workaround for https://github.com/quarkusio/quarkus/issues/7862 (this issue is closed, but having optional nested in other optional objects is not working).